Ensure the Nebula device exists before attempting to bind to the Nebula IP

brad-defined opened the pull request:
As reported in https://github.com/slackhq/nebula/issues/243 and https://github.com/slackhq/nebula/issues/314, internal services that are configured to bind only to the Nebula IP sometimes fail because they attempt to bind to the Nebula interface before it's created. This diff moves the start of the SSH, DNS, and Stats internal services until after we're certain that the Nebula interface is created.

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/slackhq/nebula?pullRequest=375) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you sign our [Contributor License Agreement](https://cla-assistant.io/slackhq/nebula?pullRequest=375) before we can accept your contribution.<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/slackhq/nebula?pullRequest=375) it.</sub>

brad-defined commented:
I rebased against master, and re-tested. I discovered that I never actually fixed the race in the stats case, so that code moved around a little bit. (I didn't wrap the Listen call in the delayed function, so it still failed when bound only to the Nebula IP. Oops.)

I also updated the Control struct to have named variables for the delayed startup functions, instead of the more generic array.

nbrownus commented on sshd/server.go:
```diff
@@ -93,6 +93,7 @@ func (s *SSHServer) Run(addr string) error {
 	var err error
 	s.listener, err = net.Listen("tcp", addr)
 	if err != nil {
+		s.l.WithField("err", err).Warn("Error starting SSH server listener, shutting down")
```
Caller should be the one to log

brad-defined replied:
Moved the log line, and verified that I saw the error log when misconfiguring the ssh listener.

nbrownus commented on stats.go:
```diff
@@ -17,24 +17,35 @@ import (
 
+	var startFn func()
 	switch mType {
 	case "graphite":
```
I recall an old debate regarding this, I think, will need to noodle

nbrownus reviewed (approved):
Looks good, also closes #393
